oidc-api/src/main/java/uk/gov/di/authentication/oidc/lambda/LogoutHandler.java

```diff
@@ -92,170 +92,147 @@ public APIGatewayProxyResponseEvent handleRequest(
-                cookieHelper.parseSessionCookie(input.getHeaders()).orElseThrow();
-
-        attachSessionIdToLogs(session);
-        attachLogFieldToLogs(CLIENT_SESSION_ID, sessionCookieIds.getClientSessionId());
-        attachLogFieldToLogs(GOVUK_SIGNIN_JOURNEY_ID, sessionCookieIds.getClientSessionId());
-
-        LOG.info("LogoutHandler processing request");
-
-        if (!session.getClientSessions().contains(sessionCookieIds.getClientSessionId())) {
-            LOG.warn("Client Session ID does not exist");
-            return logoutService.generateErrorLogoutResponse(
-                    Optional.empty(),
-                    new ErrorObject(OAuth2Error.INVALID_REQUEST_CODE, "invalid session"),
-                    input,
-                    Optional.empty(),
-                    Optional.of(session.getSessionId()));
-        }
-
-        Map<String, String> queryStringParameters = input.getQueryStringParameters();
-        if (queryStringParameters == null || queryStringParameters.isEmpty()) {
-            LOG.info("Deleting session and returning default logout as no input parameters");
-            segmentedFunctionCall("destroySessions", () -> logoutService.destroySessions(session));
-            return logoutService.generateDefaultLogoutResponse(
-                    state, input, Optional.empty(), Optional.of(session.getSessionId()));
+            state = Optional.ofNullable(queryStringParameters.get("state"));
         }
 
         Optional<String> idTokenHint =
                 Optional.ofNullable(queryStringParameters.get("id_token_hint"));
+        Optional<String> audience;
         Optional<String> postLogoutRedirectUri =
                 Optional.ofNullable(queryStringParameters.get("post_logout_redirect_uri"));
-        Optional<String> audience = Optional.empty();
-
         if (idTokenHint.isPresent()) {
             LOG.info("ID token hint is present");
-            if (!doesIDTokenExistInSession(idTokenHint.get(), session)) {
-                LOG.warn("ID token does not exist");
-                return unableToValidateIdTokenHint(input, session);
-            }
             boolean isTokenSignatureValid =
                     segmentedFunctionCall(
                             "isTokenSignatureValid",
                             () -> tokenValidationService.isTokenSignatureValid(idTokenHint.get()));
             if (!isTokenSignatureValid) {
                 LOG.warn("Unable to validate ID token signature");
-                return unableToValidateIdTokenHint(input, session);
+                return logoutService.generateErrorLogoutResponse(
+                        Optional.empty(),
+                        new ErrorObject(
+                                OAuth2Error.INVALID_REQUEST_CODE,
+                                "unable to validate id_token_hint"),
+                        input,
+                        Optional.empty(),
+                        Optional.empty());
```

Comment:
Please could you be able to make the commits a bit more readable (this was changed in the last commit and now it is being reverted?) and then I will re-review

Reply:
Still a bit confusing but no strange change flows